zyte_spider_templates/pages/search_request_template.py

```diff
@@ -0,0 +1,326 @@
+    async def to_item(self) -> SearchRequestTemplate:
+        builders = {
+            "extruct": self._item_from_extruct,
+            "formasaurus": self._item_from_formasaurus,
+            "link_heuristics": self._item_from_link_heuristics,
+            "form_heuristics": self._item_from_form_heuristics,
+        }
+        builder_ids = self.page_params.get("search_request_builders", list(builders))
+        builder_strategy = self.page_params.get(
+            "search_request_builder_strategy", "popular"
+        )
+        if builder_strategy not in {"first", "popular"}:
+            raise ValueError(
+                f"Unsupported search_request_builder_strategy value: {builder_strategy!r}"
+            )
+        results = defaultdict(list)
+        for builder_id in builder_ids:
+            builder = builders[builder_id]
+            try:
+                result = builder()
+            except ValueError:
+                continue
+            if result:
+                if builder_strategy == "first":
+                    return result
+                results[(result.url, result.body)].append((builder_id, result))
+        if results:
+            assert builder_strategy == "popular"
+            top_count = max(len(v) for v in results.values())
+            top_results = {
+                builder_id: result
+                for result_list in results.values()
+                for builder_id, result in result_list
+                if len(result_list) == top_count
+            }
+            for builder_id in builder_ids:
+                if builder_id not in top_results:
+                    continue
+                return top_results[builder_id]
+
+        if not isinstance(self.response.response, BrowserResponse):
+            raise UseFallback(
+                f"Cannot build a search request template for "
+                f"{self.response.url}. Try enabling browser rendering by "
+                f'setting "browserHtml" to True in the '
+                f"ZYTE_API_PROVIDER_PARAMS setting dict."
+            )
+
+        raise UseFallback(
+            f"Cannot build a search request template for "
+            f"{self.response.url}. A quick workaround would be to use a "
+            f"search URL as input URL instead of using the search "
+            f"queries input field. You can also manually implement "
+            f"search support for a given website "
+            f"(https://zyte-common-items.readthedocs.io/en/latest/usage/request-templates.html#writing-a-request-template-page-object). "
+            f"You can also submit a request to add support for a specific "
+            f"website in the public GitHub issue tracker "
+            f"(https://github.com/zytedata/zyte-spider-templates/issues) "
+            f"or open a private support ticket if you are a Zyte customer "
+            f"(https://support.zyte.com/support/tickets/new), but there "
+            f"is no guarantee that we will implement support for it."
```

Comment:
I'd drop it; not sure if we want to keep website-specific search templates in this repo (at least not now probably).